orangecontrib/text/widgets/ownyt.py

```diff
@@ -242,88 +265,92 @@ def retrieve_remaining_records(self):
-            self.run_query_button.setEnabled(False)
-
-            # Accumulate remaining results in these lists.
-            remaining_docs = []
-            num_metas = len(self.output_corpus.domain.metas)
-            remaining_metas = np.empty((0, num_metas), dtype=object)
-            remaining_classes = []
-
-            self.query_running = True
-            self.progressBarInit()
-            for i in range(int(self.num_retrieved/10), num_steps):
-                # Stop querying if the flag is not set.
-                if not self.query_running:
-                    break
-
-                # Update the progress bar.
-                self.progressBarSet(100.0 * (i/num_steps))
-
-                res, cached, error = self.nyt_api.execute_query(i)
-
-                if res:
-                    docs, metas, meta_vars, class_values = parse_record_json(res, self.nyt_api.includes_fields)
-                    remaining_docs += docs
-                    remaining_metas = np.vstack((remaining_metas, np.array(metas)))
-                    remaining_classes += class_values
-
-                    # Update the info label.
-                    self.num_retrieved += len(res["response"]["docs"])
-                    info_label = "Records: {}\nRetrieved: {}".format(self.all_hits, self.num_retrieved)
-                    if self.all_hits > 1000:
-                        info_label += " (max 1000)"
-                    self.query_info_label.setText(info_label)
-
-                    if not cached:  # Only wait if an actual request was made.
-                        sleep(1)
-                else:
-                    if error:
-                        if isinstance(error, HTTPError):
-                            self.error(1, "An error occurred(HTTP {})".format(error.code))
-                        elif isinstance(error, URLError):
-                            self.error(1, "An error occurred(URL {})".format(error.reason))
-                        break
-            self.progressBarFinished()
-            self.query_running = False
+        if not self.nyt_api:
+            return
+
+        num_steps = min(math.ceil(self.all_hits/10), 100)
+
+        # Update buttons.
+        self.retrieve_other_button.setText('Stop retrieving')
+        self.open_set_api_key_dialog_button.setEnabled(False)
+        self.run_query_button.setEnabled(False)
+
+        self.query_running = True
+        self.progressBarInit()
+        for i in range(int(self.num_retrieved/10), num_steps):
+            # Stop querying if the flag is not set.
+            if not self.query_running:
+                break
+
+            res, cached, error = self.nyt_api.execute_query(i)
+
+            if not res:     # Break if no response.
+                if error:
+                    self.display_error_response(error)
+                break
+
+            metas, class_values = parse_record_json(res, self.nyt_api.includes_fields)
```

Comment:
Here also we might have problems when using the scripting part only. Please, try to use the scripting part only to get the same results as in gui.